packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -155,35 +165,58 @@ export class NftDetectionController extends StaticIntervalPollingControllerV1<
 
   private getOwnerNftApi({
     address,
-    offset,
+    next,
   }: {
     address: string;
-    offset: number;
+    next?: string;
   }) {
-    return `${OPENSEA_PROXY_URL}/assets?owner=${address}&offset=${offset}&limit=50`;
+    return `${OPENSEA_PROXY_URL}/chain/${
+      OpenSeaV2ChainIds.ethereum
+    }/account/${address}/nfts?limit=200&next=${next ?? ''}`;
   }
 
   private async getOwnerNfts(address: string) {
-    let nftApiResponse: { assets: ApiNft[] };
+    let nftApiResponse: OpenSeaV2ListNftsResponse;
     let nfts: ApiNft[] = [];
-    let offset = 0;
-    let pagingFinish = false;
-    /* istanbul ignore if */
+    let next;
+
     do {
       nftApiResponse = await fetchWithErrorHandling({
-        url: this.getOwnerNftApi({ address, offset }),
+        url: this.getOwnerNftApi({ address, next }),
         timeout: 15000,
       });
 
       if (!nftApiResponse) {
         return nfts;
       }
 
-      nftApiResponse?.assets?.length !== 0
-        ? (nfts = [...nfts, ...nftApiResponse.assets])
-        : (pagingFinish = true);
-      offset += 50;
-    } while (!pagingFinish);
+      const newNfts = await Promise.all(
+        nftApiResponse.nfts.map(async (nftV2) => {
+          const nftV1 = mapOpenSeaNftV2ToV1(nftV2);
+
+          // If the image hasn't been processed into OpenSea's CDN, the image_url will be null.
+          // Try fetching the NFT individually, which returns the original image url from metadata if available.
+          if (!nftV1.image_url && nftV2.metadata_url) {
+            const nftDetails: OpenSeaV2GetNftResponse | undefined =
+              await safelyExecute(() =>
+                timeoutFetch(
+                  this.getNftApi({
+                    contractAddress: nftV2.contract,
+                    tokenId: nftV2.identifier,
+                  }),
+                  undefined,
+                  1000,
```

Comment:
In all these examples I found, OS falls back to the original, non-CDN URL when you fetch the NFT individually.  V1 also did this when listing, but in separate fields, which is how those used to render.

Alternatively we could fetch `metadata_url` and grab the image field, but was more anonying to handle ipfs, arweave.  The timeout was more for when I was doing that approach and may not be needed.

Reply:
Good call on not fetching the metadata. That would have had complicated privacy implications.